librz/analysis/esil/esil_trace.c

```diff
@@ -193,6 +225,16 @@ static int trace_hook_mem_write(RzAnalysisEsil *esil, ut64 addr, const ut8 *buf,
 	return ret;
 }
 
+/**
+ * Get instruction trace from esil trace by index
+ * \param etrace RzAnalysisEsilTrace *, esil trace
+ * \param idx int, index of instruction
+ * \return RzILTraceInstruction *, instruction trace at index
+ */
+RZ_API RzILTraceInstruction *rz_analysis_esil_get_instruction_trace(RzAnalysisEsilTrace *etrace, int idx) {
+	return rz_pvector_at(etrace->instructions, idx);
```

Comment:
Yes but you should consider the vector like a "simple array". There are checks for the indexes in rz_vector_at but they are implemented as programmer errors with rz_return_val_if_fail. Users of the API are expected to not pass wrong `idx` values to those APIs, so you should check the idx before calling vector functions.

Reply:
Great, I will add checks